File: C++/Sejal/BalancedBinaryTree.cpp
//https://www.devsnest.in/lms?id=59&slug=balanced-binary-tree&tab=question

int depth(TreeNode* root) {
    if (root == NULL)
        return 0;
    return 1 + max(depth(root->left), depth(root->right));
}
int solve(TreeNode* root){
//CODE HERE 
    int lh = depth(root->left);
    int rh = depth(root->right);
    return ((rh - lh >= -1) && (rh - lh <= 1));
}

File: java/ankit_kanyal/devnest_problems/phase_2/BalancedBinaryTree.java
// link- https://devsnest.in/algo-challenges/balanced-binary-tree?tab=question

static int solve(TreeNode root){
//CODE HERE 
if(helper(root)==-1){
    return 0;
}
else{
    return 1;
}
}

static int helper(TreeNode root){

if(root==null) return 0;
    
int lr=helper(root.left);

if(lr==-1) return -1;

int rr=helper(root.right);

if(rr==-1) return -1;
    
if(Math.abs(lr-rr)>1) return -1;
    
return Math.max(lr,rr)+1;
}